src/components/single-line-diagram.js

```diff
@@ -60,11 +61,19 @@ const SvgNotFound = (props) => {
     );
 };
 
-const nosvg = {svg: null, metadata: null, error: null, svgUrl: null};
+const noSvg = {svg: null, metadata: null, error: null, svgUrl: null};
 
-const SingleLineDiagram = (props) => {
+const SingleLineDiagram = forwardRef((props, ref)  => {
 
-    const [svg, setSvg] = useState(nosvg);
+    const [svg, setSvg] = useState(noSvg);
+    const [svgDisplayInfo, setSvgDisplayInfo] = useState();
+
+    useImperativeHandle(ref, () => ({
+        reloadSvg: (svgUrl, svgDisplayInfo) => {
+            fetchSvg(svgUrl).then(svg => setSvg(svg));
+            setSvgDisplayInfo(svgDisplayInfo);
```

Comment:
i don't think svgDisplayInfo should be a state. instead it should be a ref that doesn't trigger relayouting when it is changed.